src/components/dialogs/SpoolmanChangeSpoolDialogRow.vue

```diff
@@ -0,0 +1,105 @@
+<template>
+    <tr class="cursor-pointer" @click="setSpoolRow">
+        <td style="width: 50px" class="pr-0 py-2">
+            <spool-icon :color="color" style="width: 50px; float: left" class="mr-3" />
+        </td>
+        <td class="py-2" style="min-width: 300px">
+            <strong class="text-no-wrap">{{ vendor }} - {{ name }}</strong>
+            <template v-if="location">
+                <br />
+                {{ $t('Panels.SpoolmanPanel.Location') }}: {{ location }}
+            </template>
+            <template v-if="spool.comment">
+                <br />
+                {{ spool.comment }}
+            </template>
+        </td>
+        <td class="text-center text-no-wrap">{{ material }}</td>
+        <td class="text-right text-no-wrap">{{ last_used }}</td>
+        <td class="text-right text-no-wrap">
+            <strong>{{ rest_weight_format }}</strong>
+            <small class="ml-1">/ {{ total_weight_format }}</small>
+        </td>
+    </tr>
+</template>
+
+<script lang="ts">
+import Component from 'vue-class-component'
+import { Mixins, Prop } from 'vue-property-decorator'
+import BaseMixin from '@/components/mixins/base'
+import { ServerSpoolmanStateSpool } from '@/store/server/spoolman/types'
+@Component({})
+export default class SpoolmanChangeSpoolDialogRow extends Mixins(BaseMixin) {
+    @Prop({ required: true }) declare readonly spool: ServerSpoolmanStateSpool
+
+    get color() {
+        const color = this.spool.filament?.color_hex ?? '000'
+
+        return `#${color}`
+    }
+
+    get vendor() {
+        return this.spool.filament?.vendor?.name ?? 'Unknown'
+    }
+
+    get name() {
+        return this.spool.filament?.name ?? 'Unknown'
+    }
+
+    get location() {
+        return this.spool.location
+    }
+
+    get material() {
+        return this.spool.filament?.material ?? '--'
+    }
+
+    get used_weight() {
+        return this.spool.used_weight ?? 0
+    }
+
+    get total_weight() {
+        return this.spool.filament?.weight ?? 0
+    }
+
+    get rest_weight_format() {
+        return `${(this.total_weight - this.used_weight).toFixed(0)}g`
+    }
+
+    get total_weight_format() {
```

Comment:
maybe create a mixin for weight calulations and formatting. Used in `SpoolmanPanelActiveSpool.vue`, `SpoolmanPanel.vue` and here.

Reply:
this is only needed in the `SpoolmanPanelActiveSpool.vue`.